### PR TITLE
dev: optimize to_u64_words to use span<u8> as input

### PR DESCRIPTION
**Pull request by enitrat**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe): opti

## What is the current behavior?

Resolves: #516

## What is the new behavior?

- Span<u8> are not converted to ByteArray before packing them in u64 words
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/evm/src/instructions/environmental_information.cairo [hunk 1]
```diff
@@ -240,16 +240,14 @@ impl EnvironmentInformationImpl of EnvironmentInformationTrait {
         let evm_address = self.stack.pop_eth_address()?;
 
         let account = self.state.get_account(evm_address)?;
-        if (account.is_precompile() || (account.is_ca() && account.nonce == 0)) {
+        if (account.is_precompile() || account.nonce == 0) {
```
Comment (Eikix) [inline crates/evm/src/instructions/environmental_information.cairo:243]: why is this here?

File: crates/utils/src/helpers.cairo [hunk 2]
```diff
@@ -520,13 +520,72 @@ impl SpanExtension<T, +Copy<T>, +Drop<T>> of SpanExtTrait<T> {
...
+
+        (u64_words, last_input_word, last_input_num_bytes)
+    }
+}
+
+
 #[generate_trait]
 impl BytesImpl of BytesTrait {
     // keccack256 on a bytes message
     fn compute_keccak256_hash(self: Span<u8>) -> u256 {
```
Comment (Eikix) [inline crates/utils/src/helpers.cairo:587]: it seems this trait isn't used anywhere?

Comment (enitrat) [reply crates/utils/src/helpers.cairo:587]: I think it comes from a rebase?

File: crates/evm/src/instructions/environmental_information.cairo (hunk 1 above)
Comment (enitrat) [reply crates/evm/src/instructions/environmental_information.cairo:243]: oops, idk, must have sneaked in but it was meant for another pr

**Comment by enitrat** [comment]
@Eikix rebased and ready to go 🚀

File: crates/utils/src/helpers.cairo (hunk 2 above)
Comment (Eikix) [reply crates/utils/src/helpers.cairo]: because in your pr you modify eth_transaction hashing step
maybe it's not worth it anymore to have it and could be part of this pr to delete it

or to leave it and use it anywhere we have a span of u8? 
should a span have a self method that turns these arbitrary bytes to a hash of itself?

**Review by Eikix — APPROVED** [review]
lgtm 🍀🍀

**Review by Eikix — APPROVED** [review]
lgtm
